src/js/errorMiddleware.js

```diff
@@ -0,0 +1,63 @@
+/*
+
+    A default error handler that relays an upstream error to the user. See the documentation for details:
+
+    https://github.com/GPII/gpii-express/blob/master/docs/errorMiddleware.md
+
+ */
+"use strict";
+var fluid = require("infusion");
+var gpii  = fluid.registerNamespace("gpii");
+
+fluid.registerNamespace("gpii.express.middleware.error");
+
+/**
+ *
+ * @param that - The middleware component itself.
+ * @param error - The raw upstream error.
+ * @param request {Object} The Express {Request} object.
+ * @param response {Object} The Express {Response} object.
+ * @param next {Function} The next piece of middleware in the chain.  Only used if a response has already been sent (i.e. we have no way of contacting the user).
+ *
+ */
+gpii.express.middleware.error.sendError = function (that, error, request, response, next) {
+    if (response.headersSent) {
+        return next(error);
```

Comment:
I cleaned up this cycle.  There may still be downstream middleware (logging, etc.) that can still meaningfully do its work in such a case, so the error is still passed on.  I added a fluid.fail afterward, so that the error is clearly flagged.